Sources/RInAppMessaging/Services/ConfigurationService.swift

```diff
@@ -12,20 +12,27 @@ internal enum ConfigurationServiceError: Error {
     case unknownSubscriptionId
     case invalidRequestError(UInt)
     case internalServerError(UInt)
+    case missingOrInvalidConfigURL
 }
 
 internal struct ConfigurationService: ConfigurationServiceType, HttpRequestable {
 
-    private let configURL: URL
+    private let configurationRepository: ConfigurationRepositoryType
     private(set) var httpSession: URLSession
     var bundleInfo = BundleInfo.self
 
-    init(configURL: URL, sessionConfiguration: URLSessionConfiguration) {
-        self.configURL = configURL
-        self.httpSession = URLSession(configuration: sessionConfiguration)
+    init(configurationRepository: ConfigurationRepositoryType) {
+        self.configurationRepository = configurationRepository
+        self.httpSession = URLSession(configuration: configurationRepository.defaultHttpSessionConfiguration)
     }
 
-    func getConfigData() -> Result<ConfigData, ConfigurationServiceError> {
+    func getConfigData() -> Result<ConfigEndpointData, ConfigurationServiceError> {
+        guard let configURLString = configurationRepository.getConfigEndpointURL(),
+              let configURL = URL(string: configURLString) else {
+            assertionFailure("Invalid Configuration URL: \(configurationRepository.getConfigEndpointURL() ?? "<empty>")")
```

Comment:
Is `assertionFailure` necessary here as `.failure(.missingOrInvalidConfigURL)` is returned just after?